lib/ide/json-rpc/src/messages.rs

```diff
@@ -157,6 +157,24 @@ pub enum Result<Res> {
     Error(Error),
 }
 
+impl<Res> Result<Res> {
+    /// Construct a successful remote call result value.
+    pub fn new_success(result: Res) -> Result<Res> {
+        Result::Success(Success {result})
+    }
+
+    /// Construct a failed remote call result value.
+    pub fn new_error
+    (code:i64, message:String, data:Option<serde_json::Value>) -> Result<Res> {
+        Result::Error(Error{code,message,data})
+    }
```

Comment:
I believe these should be auto-generated with a simple macro from the definition of `Result`. Maintaining such boilerplate is painful.